File: hw/hw4_NB/Naveen_Renji_ HW_04_NB.R
# Clear environment variables
rm(list=ls())

# Import Libraries
library(tidyverse)
library(e1071)
library(caret)

# Load the dataset
df <- read.csv("breast-cancer-wisconsin.csv")

# Print the first few rows
head(df)

# Remove the first column as it is not needed
df <- df[,-1]

# Taking care of missing values
df <- df %>%
  mutate(F6 = ifelse(F6 == "?", NA, as.numeric(F6))) %>%
  drop_na()

# Split the data into training and test sets
set.seed(123) # For reproducibility
trainIndex <- createDataPartition(df$Class, p = .7, list = FALSE, times = 1)
train <- df[trainIndex, ]
test <- df[-trainIndex, ]

# Create NB Model and classify the training data
y_train <- ifelse(train$Class == 2, 0, 1)
y_test <- ifelse(test$Class == 2, 0, 1)
X_train <- train %>% select(-Class) %>% as.matrix()
X_test <- test %>% select(-Class) %>% as.matrix()

# Gaussian Naive Bayes
nbg <- naiveBayes(x = X_train, y = y_train, type = "raw")
Y_pred <- predict(nbg, X_test, type = "raw")
confusionMatrix(Y_pred, y_test, positive = "0")
confusionMatrix(Y_pred, y_test, positive = "0")$table

# Bernoulli Naive Bayes
nbb <- naiveBayes(x = X_train, y = y_train, type = "class")
Y_pred <- predict(nbb, X_test, type = "class")
confusionMatrix(Y_pred, y_test, positive = "0")
confusionMatrix(Y_pred, y_test, positive = "0")$table

# Multinomial Naive Bayes
nbm <- naiveBayes(x = X_train, y = y_train, type = "count")
Y_pred <- predict(nbm, X_test, type = "raw")
confusionMatrix(Y_pred, y_test, positive = "0")
confusionMatrix(Y_pred, y_test, positive = "0")$table

File: hw/hw4_NB/Naveen_Renji_ HW_04_NB.py
#!/usr/bin/env python
# coding: utf-8

# # Name : Naveen Mathews Renji | CWID : 20016323 

# ## Import Libraries

# In[1]:


import pandas as pd
import numpy as np 
import seaborn as sns
import matplotlib.pyplot as plt
from pandas.plotting import scatter_matrix
from sklearn.naive_bayes import GaussianNB
from sklearn.naive_bayes import MultinomialNB
from sklearn.naive_bayes import BernoulliNB
from sklearn.model_selection import train_test_split
from sklearn.metrics import classification_report, confusion_matrix


# ## Load the dataset

# In[2]:


df = pd.read_csv("breast-cancer-wisconsin.csv")


# In[3]:


df.head()


# In[4]:


num_rows = df.shape[0]
print(num_rows)


# ## Taking care of missing values

# In[5]:


df['F6']=df['F6'].replace('?',np.nan)
df = df.dropna()
del df['Sample']


# In[6]:


df.describe()


# ## Creating NB Model and classifying the training data

# In[7]:


y=df['Class']
y=np.where(y==2,0,1)
X=df.drop('Class',axis=1).values


# In[8]:


X_train,X_test,y_train,y_test=train_test_split(X,y,test_size=0.30)


# ## Guassian Naive Bayes

# In[9]:


nbg = GaussianNB()
nbg.fit(X_train, y_train)
nbg.score(X_test, y_test)
Y_pred=nbg.predict(X_test)
print("The Confusion Matrix is -")
print(confusion_matrix(y_test,Y_pred))
print("The Classification Report Matrix is -")
print(classification_report(y_test,Y_pred))


# ## Bernoulli Naive Bayes

# In[10]:


nbb = BernoulliNB()
nbb.fit(X_train, y_train)
nbb.score(X_test, y_test)
Y_pred=nbb.predict(X_test)
print("The Confusion Matrix is -")
print(confusion_matrix(y_test,Y_pred))
print("The Classification Report Matrix is -")
print(classification_report(y_test,Y_pred))


# ## Multinomial Naive Bayes

# In[11]:


nbm = MultinomialNB()
nbm.fit(X_train, y_train)
nbm.score(X_test, y_test)
Y_pred=nbm.predict(X_test)
print("The Confusion Matrix is -")
print(confusion_matrix(y_test,Y_pred))
print("The Classification Report Matrix is -")
print(classification_report(y_test,Y_pred))


# In[ ]:




